src/klein/_plating.py

```diff
@@ -0,0 +1,149 @@
+# -*- test-case-name: klein.test.test_plating -*-
+
+"""
+Templating wrapper support for Klein.
+"""
+
+from functools import wraps
+from json import dumps
+
+from six import text_type, integer_types
+
+from twisted.web.template import TagLoader, Element
+from twisted.web.error import MissingRenderMethod
+
+def _should_return_json(request):
+    """
+    Should the given request result in a JSON entity-body?
+    """
+    return bool(request.args.get(b"json"))
+
+
+def json_serialize(item):
+    """
+    A function similar to L{dumps}.
+    """
+    def helper(unknown):
+        if isinstance(unknown, PlatedElement):
+            return unknown.slot_data
+        else:
+            raise TypeError("{input} not JSON serializable"
+                            .format(input=unknown))
+    return dumps(item, default=helper)
+
+
+def _extra_types(input):
+    """
+    Renderability for a few additional types.
+    """
+    if isinstance(input, (float,) + integer_types):
+        return text_type(input)
+    return input
+
+
+class PlatedElement(Element):
+    """
+    The element type returned by L{Plating}.  This contains several utility
+    renderers.
+    """
+
+    def __init__(self, slot_data, preloaded):
+        """
+        @param slot_data: A dictionary mapping names to values.
+
+        @param preloaded: The pre-loaded data.
+        """
+        self.slot_data = slot_data
+        super(PlatedElement, self).__init__(
+            loader=TagLoader(preloaded.fillSlots(
+                **{k: _extra_types(v) for k, v in slot_data.items()}
+            ))
+        )
+
+    def lookupRenderMethod(self, name):
+        """
+        @return: a renderer.
+        """
+        if ":" not in name:
+            raise MissingRenderMethod(self, name)
+        slot, type = name.split(":", 1)
+
+        def renderList(request, tag):
+            for item in self.slot_data[slot]:
+                yield tag.fillSlots(item=_extra_types(item))
+        types = {
+            "list": renderList,
+        }
+        if type in types:
+            return types[type]
+        else:
+            raise MissingRenderMethod(self, name)
+
+
+class Plating(object):
+    """
+    A L{Plating} is a container which can be used to generate HTML from data.
+
+    Its name is derived both from tem-I{plating} and I{chrome plating}.
+    """
+
+    CONTENT = "klein:plating:content"
+
+    def __init__(self, defaults=None, tags=None,
+                 presentation_slots=frozenset()):
+        """
```

Comment:
`presentation_slots` is a good, descriptive name, but it'd be nice to see the docstring from [the test](https://github.com/twisted/klein/pull/118/files#diff-498458908842b5b11e2dbbf15a52d261R202) in the initializer's docstring.   It doesn't seem like Klein follows the same documentation standard as say, Twisted, so this comment is just advisory and doesn't need to be addressed for a merge.